source/common.c

```diff
@@ -187,44 +217,127 @@ static struct aws_error_info errors[] = {
-    AWS_DEFINE_ERROR_INFO(AWS_ERROR_MUTEX_CALLER_NOT_OWNER, "the caller of a mutex operation was not the owner", AWS_LIB_NAME),
-    AWS_DEFINE_ERROR_INFO(AWS_ERROR_MUTEX_FAILED, "mutex operation failed", AWS_LIB_NAME),
-    AWS_DEFINE_ERROR_INFO(AWS_ERROR_COND_VARIABLE_INIT_FAILED, "condition variable initialization failed.", AWS_LIB_NAME),
-    AWS_DEFINE_ERROR_INFO(AWS_ERROR_COND_VARIABLE_TIMED_OUT, "condition variable wait timed out.", AWS_LIB_NAME),
-    AWS_DEFINE_ERROR_INFO(AWS_ERROR_COND_VARIABLE_ERROR_UNKNOWN, "condition variable unknown error.", AWS_LIB_NAME),
-    AWS_DEFINE_ERROR_INFO(AWS_ERROR_CLOCK_FAILURE, "clock get ticks operation failed", AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_SHORT_BUFFER,
+        "Insufficient data in input buffer",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_OVERFLOW_DETECTED,
+        "fixed size value overflow was detected",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_INVALID_BUFFER_SIZE,
+        "invalid buffer size",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_INVALID_HEX_STR,
+        "invalid hex string",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_INVALID_BASE64_STR,
+        "invalid base64 string",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_INVALID_INDEX,
+        "invalid index for list access",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_THREAD_INVALID_SETTINGS,
+        "invalid thread settings",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_THREAD_INSUFFICIENT_RESOURCE,
+        "thread, insufficient resources",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_THREAD_NO_PERMISSIONS,
+        "insufficient permissions for thread operation",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_THREAD_NOT_JOINABLE,
+        "thread not join-able",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_THREAD_NO_SUCH_THREAD_ID,
+        "no such thread id",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_THREAD_DEADLOCK_DETECTED,
+        "deadlock detected",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_MUTEX_NOT_INIT,
+        "mutex not initialized",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_MUTEX_TIMEOUT,
+        "mutex operation timed out",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_MUTEX_CALLER_NOT_OWNER,
+        "the caller of a mutex operation was not the owner",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_MUTEX_FAILED,
+        "mutex operation failed",
+        AWS_LIB_NAME),
+    AWS_DEFINE_ERROR_INFO(
+        AWS_ERROR_COND_VARIABLE_INIT_FAILED,
+        "condition variable initialization failed.",
+        AWS_LIB_NAME),
```

Comment:
[pretty please] we could eliminate all those `AWS_LIB_NAME),` lines with a local macro:
```
#define AWS_DEFINE_ERROR_INFO_COMMON(code, msg) \
    AWS_DEFINE_ERROR_INFO((code), (msg), AWS_LIB_NAME)
```
remember this morning when you taught me what currying meant